### PR TITLE
[msbuild] Add JavaSourceJar build action to Binding project.

### PR DESCRIPTION
**Pull request by atsushieno**
JavaSourceJar works like JavaDocJar, but in slightly more complicated way.

JavaDocJar only extracts the jar contents and pass index.html as JavaDocIndex,
while JavaSourceJar extracts the jar, then run `javadoc` to generate javadoc,
then pass its generated index.html as JavaDocIndex.

### CODE REVIEW

**Comment by atsushieno** [comment]
Test to add:

	  <ItemGroup>
		<LibraryProjectZip Include="Jars\runtime-1.0.3.aar" />
		<EmbeddedJar Include="Jars\common-1.0.0.jar" />
		<EmbeddedJar Include="Jars\common-1.0.3.jar" />
		<JavaSourceJar Include="Jars\common-1.0.0-sources.jar" />
		<JavaSourceJar Include="Jars\runtime-1.0.3-sources.jar" />
		<JavaSourceJar Include="Jars\common-1.0.3-sources.jar" />
	  </ItemGroup>

... and ...

	<metadata>
		<attr path="/api/package[@name='android.arch.lifecycle']" name="managedName">Android.Arch.Lifecycles</attr>
		<attr path="/api/package[@name='android.arch.lifecycle']/interface[@name='LifecycleRegistryOwner']/method[@name='getLifecycle']" name="managedReturn">Android.Arch.Lifecycles.Lifecycle</attr>
	</metadata>

**Comment by jonpryor** [comment]
Could you please update `Documentation/build_process.md` to document the `@(JavaSourceJar)` build action, and how it should be used?

A test of some form would also be useful, either within `src/Xamarin.Android.Build.Tasks/Tests` or `tests/CodeGen-Binding`.

**Comment by jonpryor** [comment]
build

**Comment by atsushieno** [comment]
Currently it is stuck because Xamarin.Android.sln is broken enough to run NUnit tests on MonoDevelop (on Linux).

**Comment by atsushieno** [comment]
Tests are added.

It seems that `Documentation/build_process.md` does not contain ANY item for binding projects. If it is being added it should be in separate PR.

File: src/Xamarin.Android.Build.Tasks/Tasks/JavaDoc.cs [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+using Xamarin.Android.Tools.Aidl;
+
+namespace Xamarin.Android.Tasks
+{
+
+	public class JavaDoc : JavaToolTask
+	{
+		public string [] SourceDirectories { get; set; }
+
+		[Output]
```
Comment (jonpryor) [inline src/Xamarin.Android.Build.Tasks/Tasks/JavaDoc.cs]: The `[Output]` custom attribute is for use with the `<Output/>` element, which isn't being used. `[Output]` isn't needed.

**Pull request merged**